VimeoUpload/Upload/Networking/New Upload (Private)/VimeoSessionManager+Upload.swift

```diff
@@ -38,70 +38,58 @@ public typealias UploadParameters = [String: Any]
+                        completionHandler(nil, error as NSError)
+                    }
                 }
             })
-        })
-        
-        task.taskDescription = UploadTaskDescription.CreateVideo.rawValue
-        
-        return task
+        }
+
     }
     
-    func createVideoDownloadTask(url: URL, videoSettings: VideoSettings?, uploadParameters: UploadParameters) throws -> URLSessionDownloadTask
+    func createVideoDownloadTask(url: URL, videoSettings: VideoSettings?, uploadParameters: UploadParameters) throws -> Task?
     {
-        let request = try (self.requestSerializer as! VimeoRequestSerializer).createVideoRequest(with: url, videoSettings: videoSettings, uploadParameters: uploadParameters)
-        
-        let task = self.downloadTask(with: request as URLRequest, progress: nil, destination: nil, completionHandler: nil)
-        
-        task.taskDescription = UploadTaskDescription.CreateVideo.rawValue
-        
+        let request = try self.jsonRequestSerializer.createVideoRequest(with: url, videoSettings: videoSettings, uploadParameters: uploadParameters) as URLRequest        
+        let task = self.download(request) { _ in }
         return task
     }
     
-    func uploadVideoTask(source: URL, request: URLRequest, completionHandler: ErrorBlock?) -> URLSessionUploadTask
+    func uploadVideoTask(source: URL, request: URLRequest, completionHandler: ErrorBlock?) -> Task?
     {
-        let task = self.uploadTask(with: request, fromFile: source, progress: nil, completionHandler: { [weak self] (response, responseObject, error) -> Void in
-            
-            guard let strongSelf = self, let completionHandler = completionHandler else
-            {
-                return
-            }
-            
-            do
-            {
-                try (strongSelf.responseSerializer as! VimeoResponseSerializer).process(uploadVideoResponse: response, responseObject: responseObject as AnyObject?, error: error as NSError?)
-                completionHandler(nil)
-            }
-            catch let error as NSError
-            {
-                completionHandler(error)
+        return self.upload(request, sourceFile: source) { [jsonResponseSerializer] sessionManagingResult in
+            switch sessionManagingResult.result {
+            case .failure(let error):
+                completionHandler?(error as NSError)
+            case .success(let json):
+                do {
+                    try jsonResponseSerializer.process(
+                        uploadVideoResponse: sessionManagingResult.response,
+                        responseObject: json as AnyObject,
+                        error: nil
+                    )
+                    completionHandler?(nil)
+                } catch {
+                    completionHandler?(error as NSError)
+                }
             }
-            
-        })
-        
-        task.taskDescription = UploadTaskDescription.UploadVideo.rawValue
```

Comment:
Double checking if we were using this before - but looks like we're not using `UploadTaskDescription`

Reply:
Good question! And the answer is no, it is not used anywhere. We do use the taskId but the descriptor is simply ignored.